### PR TITLE
Add Lowest common ancestor to RMQ-plus minus 1

### PR DESCRIPTION
**Pull request by minchopaskal**
* moved rmq.{hpp,cpp} to their own folder
* implemented lca2rmq algorithm

### CODE REVIEW

File: search/LowestCommonAncesstor2RMQ/rmq_pm1.cpp [hunk 1]
```diff
@@ -0,0 +1,205 @@
+#include "rmq_pm1.hpp"
+
+using baron::RMQPM1;
```
Comment (GeorgiSamardzhiev) [inline search/LowestCommonAncesstor2RMQ/rmq_pm1.cpp]: Not sure if such nesting is needed

**Comment by GeorgiSamardzhiev** [comment]
You do not need the extra folders if Tree class is created in the graph folder

File: search/LowestCommonAncesstor2RMQ/rmq_pm1.cpp (hunk 1 above)
Comment (minchopaskal) [reply search/LowestCommonAncesstor2RMQ/rmq_pm1.cpp]: Okay, I've also namespaced the RMQ class with "baron". I presume we won't use a namespace for the library?

Edit: I see the baron namespace is used elsewhere, so is the folder structure the thing you are referring to?

**Comment by minchopaskal** [comment]
Removed the added folder hierarchy, fixed a small bug.

**Pull request merged**